In [34]:
import numpy as np
import pandas as pd
from pathlib import Path
from sklearn.model_selection import train_test_split

## Retrieve the data

In [35]:
# Import the data
df=pd.read_csv('Resources/lending_data.csv')
df.head()

loan_size  interest_rate  borrower_income  debt_to_income  num_of_accounts  \
0    10700.0          7.672            52800        0.431818                5   
1     8400.0          6.692            43600        0.311927                3   
2     9000.0          6.963            46100        0.349241                3   
3    10700.0          7.664            52700        0.430740                5   
4    10800.0          7.698            53000        0.433962                5   

   derogatory_marks  total_debt  loan_status  
0                 1       22800            0  
1                 0       13600            0  
2                 0       16100            0  
3                 1       22700            0  
4                 1       23000            0

In [36]:
# Set target, features and feature_names.
target = df["loan_status"]
data = df.drop("loan_status", axis=1)
feature_names = data.columns
data.head()

loan_size  interest_rate  borrower_income  debt_to_income  num_of_accounts  \
0    10700.0          7.672            52800        0.431818                5   
1     8400.0          6.692            43600        0.311927                3   
2     9000.0          6.963            46100        0.349241                3   
3    10700.0          7.664            52700        0.430740                5   
4    10800.0          7.698            53000        0.433962                5   

   derogatory_marks  total_debt  
0                 1       22800  
1                 0       13600  
2                 0       16100  
3                 1       22700  
4                 1       23000

## Consider the models
  * Prediction: My prediction is logistic regression will perform better because the dataset is all numeric, Random Forest Classifier performs better with more categorical data than numeric.

## Fit a LogisticRegression model and RandomForestClassifier model

In [37]:
# Split the data into X_train, X_test, y_train, y_test
X_train, X_test, y_train, y_test = train_test_split(data, target, random_state=1)

In [38]:
from sklearn.preprocessing import StandardScaler
scaler = StandardScaler().fit(X_train)
X_train_scaled = scaler.transform(X_train)
X_test_scaled = scaler.transform(X_test)

In [39]:
#Create a logistic regression model
from sklearn.linear_model import LogisticRegression
classifier = LogisticRegression()

# Train a Logistic Regression model and print the model score
#Fit (train) our model by using the training data
classifier.fit(X_train_scaled, y_train)
print(f"Training Data Score: {classifier.score(X_train_scaled, y_train):.3f}")
print(f"Testing Data Score: {classifier.score(X_test_scaled, y_test):.3f}")


Training Data Score: 0.994
Testing Data Score: 0.994


In [40]:
# Train a Random Forest Classifier model and print the model score
from sklearn.ensemble import RandomForestClassifier
clf = RandomForestClassifier(random_state=1, n_estimators=50).fit(X_train_scaled, y_train)
print(f'Training Score: {clf.score(X_train_scaled, y_train):.3f}')
print(f'Testing Score: {clf.score(X_test_scaled, y_test):.3f}')

Training Score: 0.997
Testing Score: 0.991


## Conclusion
  Which model performed better? How does that compare to your prediction? 
 
  Both models perform well with the dataset and have the similar scores. My prediction was correct because the Testing Score of logistic regression (0.994) which is slightly higher than Random Forest Classifier (0.991).
 